src/arcaflow_plugin_sdk/schema.py

```diff
@@ -5989,10 +5989,17 @@ def __call__(
             input.validate(params, tuple(["input"]))
         # Run the step
         result = self._handler(step_local_data.initialized_object, params)
+        if not isinstance(result, tuple):
+            raise BadArgumentException(
+                f"The implementation of step {run_id}/{self.id} returned"
+                f" type {type(result)}; expected a tuple with two"
+                " values: output ID string and a step-specific value."
+            )
         if len(result) != 2:
             raise BadArgumentException(
-                "The step returned {} results instead of 2. Did your step"
-                " return the correct results?".format(len(result))
+                f"The implementation of step {run_id}/{self.id} returned"
+                f"{len(result)} results instead of 2. Did your step"
```

Comment:
"Did your step return the correct results" is a difficult message to evaluate without knowing *what* was returned, and printing out a potentially long value may not be practical here ... but I know that if I got this I'd be cursing the system for complaining and then deliberately withholding relevant information it clearly possesses. 😆

Reply:
I'll print the result.